capstan_test.go

```diff
@@ -22,31 +22,38 @@ func runCapstan(command []string, root string) *exec.Cmd {
 	return c
 }
 
-func TestCommandErrors(t *testing.T) {
+func TestCommandErrorCodes(t *testing.T) {
 	root, err := ioutil.TempDir("", "capstan-root")
 	if err != nil {
 		t.Errorf("capstan: %v", err)
 	}
 	defer os.RemoveAll(root)
 
-	m := map[string]string{
-		"build foo":  "open Capstanfile: no such file or directory\n",
-		"build":      "usage: capstan build [image-name]\n",
-		"pull":       "usage: capstan pull [image-name]\n",
-		"import":     "usage: capstan import [image-name] [image-file]\n",
-		"import foo": "usage: capstan import [image-name] [image-file]\n",
-		"rmi":        "usage: capstan rmi [image-name]\n",
-		"run foo":    "foo: no such image\n",
-		"run":        "No Capstanfile found, unable to run.\n",
-	}
-	for key, value := range m {
-		cmd := runCapstan(strings.Fields(key), root)
-		out, err := cmd.Output()
-		if err != nil {
-			t.Errorf("capstan: %v", err)
+	m := []struct {
+		cmd     string
+		errCode int
+		errMsg  string
+	}{
+		{"build foo", 65, "open Capstanfile: no such file or directory\n"},
+		{"build", 64, "usage: capstan build [image-name]\n"},
+		{"pull", 64, "usage: capstan pull [image-name]\n"},
+		{"import", 64, "usage: capstan import [image-name] [image-file]\n"},
+		{"import foo", 64, "usage: capstan import [image-name] [image-file]\n"},
+		{"rmi", 64, "usage: capstan rmi [image-name]\n"},
+		{"run foo", 65, "foo: no such image\n"},
+		{"run", 65, "Missing Capstanfile or package metadata\n"},
+		{"package help compose", 0, "capstan package compose - composes the package and all its dependencies into OSv imag"},
+	}
+	for _, args := range m {
+		cmd := runCapstan(strings.Fields(args.cmd), root)
```

Comment:
If I remember correctly, some of the tests need to know where the root is. Also, we should not `defer os.RemoveAll(root) in the `runCapstan` as it will remove the root repository as soon as the `runCapstan` finishes.